scripts/o3de/o3de/compatibility.py

```diff
@@ -0,0 +1,295 @@
+#
+# Copyright (c) Contributors to the Open 3D Engine Project.
+# For complete copyright and license terms please see the LICENSE at the root of this distribution.
+#
+# SPDX-License-Identifier: Apache-2.0 OR MIT
+#
+#
+"""
+This file contains version utilities
+"""
+from packaging.version import Version, InvalidVersion
+from packaging.specifiers import SpecifierSet
+import pathlib
+import logging
+from o3de import manifest, utils, cmake
+
+logger = logging.getLogger('o3de.compatibility')
+logging.basicConfig(format=utils.LOG_FORMAT)
+
+def project_engine_compatible(project_path:pathlib.Path, project_json_data:dict, engine_json_data:dict) -> bool:
+    """
+    Returns True if the project is compatible with the provided engine data.
+    :param project_path: path to the project
+    :param project_json_data: project json data dictionary
+    :param engine_json_data: engine json data dictionary
+    """
+    if not o3de_object_is_compatible_with_engine(project_json_data, engine_json_data):
+        logger.error(f'{project_json_data["project_name"]} is not known to be compatible with the '
+                        f'engine {engine_json_data["engine_name"]}{engine_json_data.get("version","")} and requires the --force parameter '
+                        'to register.')
+        return False
+
+    # verify project -> gem -> engine compatibility
+    enabled_gems_file = cmake.get_enabled_gem_cmake_file(project_path=project_path)
+    gem_names = cmake.get_enabled_gems(enabled_gems_file)
+
+    # it's much more efficient to get all gem data once than to query them by name one by one
+    gem_paths = manifest.get_all_gems(project_path)
+    gems_json_data = {}
+    for gem_path in gem_paths:
+        gem_path = pathlib.Path(gem_path).resolve()
```

Comment:
The gem_paths should already be [resolved](https://github.com/o3de/o3de/blob/development/scripts/o3de/o3de/manifest.py#L219-L221) .
So the call `.resolve` isn't needed.
```suggestion
        gem_path = pathlib.Path(gem_path)
```